### DISCUSSION TITLE
Upgrade on different computer than where Install was executed

### DISCUSSION THREAD

**Discussion by DevOpsDrum** [getporter/porter#1976] [Q&A]
## Issue
I'm creating a pipeline to update images in an existing environment. My assumption is that using the `porter upgrade` command is the proper way to handle this, but I get an error "could not load installation my-project: Installation does not exist". 

## Question
Is there a procedure that would allow me to run `porter upgrade` on a computer where `porter install` has not been ran before?

If needed, it would be straight forward for me to use `porter install` with an 'upgrade' parameter, as the install and upgrade steps actually call the same script and pass in an upgrade parameter which allows for bypassing install-specific steps within the script.

## Workflow
First, the `porter install` command was ran on a random cloud build agent already, and the upgrade command will run on a random cloud build agent with a clean file system.

When this pipeline runs, it will:
1. Run `porter publish --reference <image path>` to build a new bundle with the latest images and then publish the bundle to a registry
2. Run `porter upgrade my-project --reference <image path>` to apply the latest images to the K8s cluster

The logs for step 2 are:
```
upgrade my-project...
Error: could not load installation my-project: Installation does not exist. 
```

Thanks!

**Comment by carolynvs** [comment]
Porter stores information relevant to the installation that is required to run subsequent actions like upgrade later. Porter's configuration file allows you to specify a remote storage backend. You can configure this to point to the same backend on your CI servers, and then the installation's state will be available when you need to upgrade the installation.

With the v1 prerelease of Porter, the storage backend is a mongo database, https://release-v1.porter.sh/configuration/#config-file.

The stable version of Porter (v0.38) uses an older mechanism that is deprecated in favor of mongodb. I recommend using a v1 release of Porter so that you don't need to migrate your data when 1.0 comes out.
